### PR TITLE
Add Dockerfiles and docker build workflow

### PR DESCRIPTION
**Pull request by weiiwang01**
The result docker images can be seen in my fork https://github.com/weiiwang01/spring-boot-sample-apps/pkgs/container/spring-boot.

### CODE REVIEW

File: spring-boot-2.7/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+RUN ./gradlew build
+
+FROM openjdk:11
+
+RUN addgroup --system spring-boot && adduser --system --no-create-home --ingroup spring-boot spring-boot
+
+USER spring-boot
+
+COPY --from=0 --chown=spring-boot:spring-boot /build/build/libs/sample-app-0.0.1-SNAPSHOT.jar /app/sample-app-0.0.1-SNAPSHOT.jar
```
Comment (gregory-schiano) [inline spring-boot-2.7/Dockerfile:15]: Probably better to name the previous stage `build` and use `--from=build` here

File: spring-boot-3.0/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+RUN ./gradlew build
+
+FROM openjdk:17
+
+RUN adduser --system --no-create-home --user-group spring-boot
+
+USER spring-boot
+
+COPY --from=0 --chown=spring-boot:spring-boot /build/build/libs/sample-app-0.0.1-SNAPSHOT.jar /app/sample-app-0.0.1-SNAPSHOT.jar
```
Comment (gregory-schiano) [inline spring-boot-3.0/Dockerfile:15]: Same remark as above

File: spring-boot-2.7/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,21 @@
+FROM gradle:jdk11
+
+COPY . /build
+
+WORKDIR /build
+
+RUN ./gradlew build
+
+FROM openjdk:11
```
Comment (gregory-schiano) [inline spring-boot-2.7/Dockerfile:9]: `openjdk` images are deprecated see
https://hub.docker.com/_/openjdk

Adopopenjdk is now part of the [Eclipse Adoptium](https://projects.eclipse.org/projects/adoptium) project and provides [Eclipse Temurin](https://adoptium.net/temurin/releases/) builds

They provide Ubuntu based images (jammy and focal
https://hub.docker.com/layers/library/eclipse-temurin/11-jre-jammy/images/sha256-772b0000ef09d2099ef07d1df87b12d942248419e79ff46fa739a5feb617e543?context=explore
https://hub.docker.com/layers/library/eclipse-temurin/17-jre-jammy/images/sha256-402c656f078bc116a6db1e2e23b08c6f4a78920a2c804ea4c2d3e197f1d6b47c?context=explore

File: spring-boot-3.0/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,21 @@
+FROM gradle:jdk17
+
+COPY . /build
+
+WORKDIR /build
+
+RUN ./gradlew build
+
+FROM openjdk:17
```
Comment (gregory-schiano) [inline spring-boot-3.0/Dockerfile:9]: Same remark as the other dockerfile

File: spring-boot-2.7/Dockerfile (hunk 3 above)
Comment (gregory-schiano) [reply spring-boot-2.7/Dockerfile:9]: Note that the gradle image is already based on Temurin Jammy images
See https://github.com/keeganwitt/docker-gradle/blob/24b8697d05b77df7eaacdbb20b8f9cc023d659fc/jdk17/Dockerfile

File: spring-boot-2.7/Dockerfile (hunk 1 above)
Comment (weiiwang01) [reply spring-boot-2.7/Dockerfile]: Updated, thanks!

File: spring-boot-3.0/Dockerfile (hunk 2 above)
Comment (weiiwang01) [reply spring-boot-3.0/Dockerfile]: Updated, thanks!

File: spring-boot-2.7/Dockerfile (hunk 3 above)
Comment (weiiwang01) [reply spring-boot-2.7/Dockerfile]: Updated, thanks!

File: spring-boot-2.7/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+USER spring-boot
+
+COPY --from=build --chown=spring-boot:spring-boot /build/build/libs/sample-app-0.0.1-SNAPSHOT.jar /app/sample-app-0.0.1-SNAPSHOT.jar
+
+WORKDIR /app
+
+HEALTHCHECK --timeout=1s CMD curl -sSf http://127.0.0.1:8080/hello-world
+
+CMD ["java", "-jar", "sample-app-0.0.1-SNAPSHOT.jar"]
```
Comment (gregory-schiano) [inline spring-boot-2.7/Dockerfile:21]: Add missing empty line ?

File: spring-boot-3.0/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+
+USER spring-boot
+
+COPY --from=build --chown=spring-boot:spring-boot /build/build/libs/sample-app-0.0.1-SNAPSHOT.jar /app/sample-app-0.0.1-SNAPSHOT.jar
+
+WORKDIR /app
+
+HEALTHCHECK --timeout=1s CMD curl -sSf http://127.0.0.1:8080/hello-world
+
+CMD ["java", "-jar", "sample-app-0.0.1-SNAPSHOT.jar"]
```
Comment (gregory-schiano) [inline spring-boot-3.0/Dockerfile:21]: Add missing empty line ?

Comment (weiiwang01) [reply spring-boot-3.0/Dockerfile]: Added, thanks!

**Pull request merged**